src/sql/parts/notebook/models/notebookContexts.ts

```diff
@@ -84,39 +88,30 @@ export class NotebookContexts {
 			return NotebookContexts.LocalContext;
 		}
 		// If no active connections exist, show "Select connection" as the default value
-		if (activeConnections.length === 0) {
+		if (activeConnections && activeConnections.length === 0) {
 			return NotebookContexts.DefaultContext;
 		}
 		// Filter active connections by their provider ids to match kernel's supported connection providers
 		else if (activeConnections.length > 0) {
 			let connections = activeConnections.filter(connection => {
 				return connProviderIds.includes(connection.providerName);
 			});
-			if (connections && connections.length > 0) {
-				defaultConnection = connections[0];
-				if (profile && profile.options) {
-					if (connections.find(connection => connection.serverName === profile.serverName)) {
-						defaultConnection = connections.find(connection => connection.serverName === profile.serverName);
-					}
-				}
-			} else if (connections.length === 0) {
+			if (connections && connections.length === 0) {
```

Comment:
This can never be undefined right? If there's no matches in the filter above it'll just be an empty array.